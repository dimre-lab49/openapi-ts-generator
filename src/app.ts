import { generateTsModels } from '.';

generateTsModels('https://im-wa-empo-nrsr.azurewebsites.net/swagger/v1/swagger.json', './output_emp/');
