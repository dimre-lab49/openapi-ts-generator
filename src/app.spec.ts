import { mkdirSync, readdirSync, readFileSync, rmdirSync, unlinkSync } from 'fs';
import { generateMessageFiles, generateUnitFiles, messageGenerationOptions, unitGenerationOptions } from './app';

describe('Full Integration Tests', () => {
  describe('MasterCorp Unit Service', () => {
    it('should generate files', async done => {
      try {
        mkdirSync(unitGenerationOptions.outputPath);
      } catch {
        // ignore
      }
      await generateUnitFiles();
      const files = readdirSync(unitGenerationOptions.outputPath).sort();
      expect(files).toMatchSnapshot();
      files.forEach(file => {
        const content = readFileSync(`${unitGenerationOptions.outputPath}${file}`, 'utf8');
        expect(content).toMatchSnapshot(file);
      });
      readdirSync(unitGenerationOptions.outputPath).forEach(file =>
        unlinkSync(`${unitGenerationOptions.outputPath}${file}`),
      );
      rmdirSync(unitGenerationOptions.outputPath);
      done();
    });
  });
  describe('MasterCorp Messaging Service', () => {
    it('should generate files', async done => {
      try {
        mkdirSync(messageGenerationOptions.outputPath);
      } catch {
        // ignore
      }
      await generateMessageFiles();
      const files = readdirSync(messageGenerationOptions.outputPath).sort();
      expect(files).toMatchSnapshot();
      files.forEach(file => {
        const content = readFileSync(`${messageGenerationOptions.outputPath}${file}`, 'utf8');
        expect(content).toMatchSnapshot(file);
      });
      readdirSync(messageGenerationOptions.outputPath).forEach(file =>
        unlinkSync(`${messageGenerationOptions.outputPath}${file}`),
      );
      rmdirSync(messageGenerationOptions.outputPath);
      done();
    });
  });
});
