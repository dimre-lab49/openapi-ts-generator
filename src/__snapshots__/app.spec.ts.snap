// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full Integration Tests MasterCorp Account Service should generate files 1`] = `
Array [
  "account-insert-request.form-group-fac.ts",
  "account-insert-request.model.ts",
  "account-insert-request.properties.ts",
  "account-update-request.form-group-fac.ts",
  "account-update-request.model.ts",
  "account-update-request.properties.ts",
  "account.form-group-fac.ts",
  "account.model.ts",
  "account.properties.ts",
  "contact-insert-request.form-group-fac.ts",
  "contact-insert-request.model.ts",
  "contact-insert-request.properties.ts",
  "contact-update-request.form-group-fac.ts",
  "contact-update-request.model.ts",
  "contact-update-request.properties.ts",
  "contact.form-group-fac.ts",
  "contact.model.ts",
  "contact.properties.ts",
  "index.ts",
]
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: account.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function AccountFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    refId: new FormControl(''),
    name: new FormControl(''),
    parentAccountId: new FormControl(''),
    isEnabled: new FormControl(''),
    emailInvoice: new FormControl(''),
    tenantId: new FormControl(''),
    parentAccount: new FormControl(''),
    contacts: new FormControl(''),
    inverseParentAccount: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: account.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { IContact } from './contact';

export interface IAccount {
  id?: string;
  refId?: string;
  name?: string;
  parentAccountId?: string;
  isEnabled?: boolean;
  emailInvoice?: boolean;
  tenantId?: string;
  parentAccount?: IAccount;
  contacts?: IContact[];
  inverseParentAccount?: IAccount[];
}
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: account.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: account-insert-request.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function AccountInsertRequestFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    refId: new FormControl(''),
    name: new FormControl('', Validators.required),
    parentAccountId: new FormControl(''),
    emailInvoice: new FormControl(''),
    contacts: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: account-insert-request.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { IContact } from './contact';

export interface IAccountInsertRequest {
  id?: string;
  refId?: string;
  name?: string;
  parentAccountId?: string;
  emailInvoice?: boolean;
  contacts?: IContact[];
}
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: account-insert-request.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: account-update-request.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function AccountUpdateRequestFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    refId: new FormControl('', Validators.required),
    name: new FormControl('', Validators.required),
    parentAccountId: new FormControl(''),
    isEnabled: new FormControl(''),
    emailInvoice: new FormControl(''),
    contacts: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: account-update-request.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { IContact } from './contact';

export interface IAccountUpdateRequest {
  id?: string;
  refId?: string;
  name?: string;
  parentAccountId?: string;
  isEnabled?: boolean;
  emailInvoice?: boolean;
  contacts?: IContact[];
}
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: account-update-request.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: contact.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function ContactFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    isEnabled: new FormControl(''),
    contactTypeId: new FormControl('', Validators.required),
    accountId: new FormControl('', Validators.required),
    firstName: new FormControl('', Validators.required),
    lastName: new FormControl('', Validators.required),
    middleName: new FormControl(''),
    addressLine1: new FormControl('', Validators.required),
    addressLine2: new FormControl(''),
    city: new FormControl('', Validators.required),
    state: new FormControl('', Validators.required),
    zip: new FormControl('', Validators.required),
    emails: new FormControl(''),
    phone1: new FormControl(''),
    phone2: new FormControl(''),
    account: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: contact.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { IAccount } from './account';

export interface IContact {
  id?: string;
  isEnabled?: boolean;
  contactTypeId?: string;
  accountId?: string;
  firstName?: string;
  lastName?: string;
  middleName?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  zip?: string;
  emails?: string;
  phone1?: string;
  phone2?: string;
  account?: IAccount;
}
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: contact.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: contact-insert-request.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function ContactInsertRequestFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    contactTypeId: new FormControl('', Validators.required),
    accountId: new FormControl('', Validators.required),
    firstName: new FormControl('', Validators.required),
    lastName: new FormControl('', Validators.required),
    middleName: new FormControl(''),
    addressLine1: new FormControl('', Validators.required),
    addressLine2: new FormControl(''),
    city: new FormControl('', Validators.required),
    state: new FormControl('', Validators.required),
    zip: new FormControl('', Validators.required),
    emails: new FormControl(''),
    phone1: new FormControl(''),
    phone2: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: contact-insert-request.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export interface IContactInsertRequest {
  id?: string;
  contactTypeId?: string;
  accountId?: string;
  firstName?: string;
  lastName?: string;
  middleName?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  zip?: string;
  emails?: string;
  phone1?: string;
  phone2?: string;
}
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: contact-insert-request.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: contact-update-request.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function ContactUpdateRequestFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl('', Validators.required),
    contactTypeId: new FormControl('', Validators.required),
    accountId: new FormControl('', Validators.required),
    firstName: new FormControl('', Validators.required),
    lastName: new FormControl('', Validators.required),
    middleName: new FormControl(''),
    addressLine1: new FormControl('', Validators.required),
    addressLine2: new FormControl(''),
    city: new FormControl('', Validators.required),
    state: new FormControl('', Validators.required),
    zip: new FormControl('', Validators.required),
    emails: new FormControl(''),
    phone1: new FormControl(''),
    phone2: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: contact-update-request.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export interface IContactUpdateRequest {
  id?: string;
  contactTypeId?: string;
  accountId?: string;
  firstName?: string;
  lastName?: string;
  middleName?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  zip?: string;
  emails?: string;
  phone1?: string;
  phone2?: string;
}
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: contact-update-request.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Account Service should generate files: index.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './account-insert-request.form-group-fac';
export * from './account-insert-request.model';
export * from './account-insert-request.properties';
export * from './account-update-request.form-group-fac';
export * from './account-update-request.model';
export * from './account-update-request.properties';
export * from './account.form-group-fac';
export * from './account.model';
export * from './account.properties';
export * from './contact-insert-request.form-group-fac';
export * from './contact-insert-request.model';
export * from './contact-insert-request.properties';
export * from './contact-update-request.form-group-fac';
export * from './contact-update-request.model';
export * from './contact-update-request.properties';
export * from './contact.form-group-fac';
export * from './contact.model';
export * from './contact.properties';
"
`;

exports[`Full Integration Tests MasterCorp Messaging Service should generate files 1`] = `
Array [
  "index.ts",
  "message.form-group-fac.ts",
  "message.model.ts",
  "message.properties.ts",
]
`;

exports[`Full Integration Tests MasterCorp Messaging Service should generate files: index.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './message.form-group-fac';
export * from './message.model';
export * from './message.properties';
"
`;

exports[`Full Integration Tests MasterCorp Messaging Service should generate files: message.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function MessageFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    recipients: new FormControl(''),
    from: new FormControl(''),
    body: new FormControl(''),
    tenantId: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Messaging Service should generate files: message.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export interface IMessage {
  id?: string;
  recipients?: string[];
  from?: string;
  body?: string;
  tenantId?: string;
}
"
`;

exports[`Full Integration Tests MasterCorp Messaging Service should generate files: message.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files 1`] = `
Array [
  "building.form-group-fac.ts",
  "building.model.ts",
  "building.properties.ts",
  "index.ts",
  "site-collection.form-group-fac.ts",
  "site-collection.model.ts",
  "site-collection.properties.ts",
  "site.form-group-fac.ts",
  "site.model.ts",
  "site.properties.ts",
  "tag.form-group-fac.ts",
  "tag.model.ts",
  "tag.properties.ts",
  "unit-tag.form-group-fac.ts",
  "unit-tag.model.ts",
  "unit-tag.properties.ts",
  "unit-type.form-group-fac.ts",
  "unit-type.model.ts",
  "unit-type.properties.ts",
  "unit.form-group-fac.ts",
  "unit.model.ts",
  "unit.properties.ts",
  "work-category-types.model.ts",
]
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: building.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function BuildingFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    siteId: new FormControl(''),
    identifier: new FormControl(''),
    isEnabled: new FormControl(''),
    site: new FormControl(''),
    units: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: building.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { ISite } from './site';
import { IUnit } from './unit';

export interface IBuilding {
  id?: string;
  siteId?: string;
  identifier?: string;
  isEnabled?: boolean;
  site?: ISite;
  units?: IUnit[];
}
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: building.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: index.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './building.form-group-fac';
export * from './building.model';
export * from './building.properties';
export * from './site-collection.form-group-fac';
export * from './site-collection.model';
export * from './site-collection.properties';
export * from './site.form-group-fac';
export * from './site.model';
export * from './site.properties';
export * from './tag.form-group-fac';
export * from './tag.model';
export * from './tag.properties';
export * from './unit-tag.form-group-fac';
export * from './unit-tag.model';
export * from './unit-tag.properties';
export * from './unit-type.form-group-fac';
export * from './unit-type.model';
export * from './unit-type.properties';
export * from './unit.form-group-fac';
export * from './unit.model';
export * from './unit.properties';
export * from './work-category-types.model';
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: site.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function SiteFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    tenantId: new FormControl(''),
    accountId: new FormControl(''),
    accountName: new FormControl(''),
    siteCollectionId: new FormControl(''),
    number: new FormControl(''),
    name: new FormControl(''),
    addressLine1: new FormControl(''),
    addressLine2: new FormControl(''),
    city: new FormControl(''),
    state: new FormControl(''),
    zip: new FormControl(''),
    isEnabled: new FormControl(''),
    siteCollection: new FormControl(''),
    buildings: new FormControl(''),
    units: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: site.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { ISiteCollection } from './site-collection';
import { IBuilding } from './building';
import { IUnit } from './unit';

export interface ISite {
  id?: string;
  tenantId?: string;
  accountId?: string;
  accountName?: string;
  siteCollectionId?: string;
  number?: string;
  name?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  zip?: string;
  isEnabled?: boolean;
  siteCollection?: ISiteCollection;
  buildings?: IBuilding[];
  units?: IUnit[];
}
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: site.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: site-collection.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function SiteCollectionFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    tenantId: new FormControl(''),
    number: new FormControl('', Validators.required),
    name: new FormControl('', Validators.required),
    code: new FormControl('', Validators.required),
    addressLine1: new FormControl(''),
    addressLine2: new FormControl(''),
    city: new FormControl(''),
    state: new FormControl('', Validators.compose([Validators.maxLength(2), Validators.minLength(2)])),
    zip: new FormControl('', Validators.compose([Validators.maxLength(10), Validators.minLength(5)])),
    isSchedulerEnabled: new FormControl('', Validators.required),
    siteCount: new FormControl(''),
    workCategoryTypes: new FormControl('', Validators.required),
    buildings: new FormControl(''),
    sites: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: site-collection.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { IWorkCategoryTypes } from './work-category-types';
import { IBuilding } from './building';
import { ISite } from './site';

export interface ISiteCollection {
  id?: string;
  tenantId?: string;
  number?: string;
  name?: string;
  code?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  zip?: string;
  isSchedulerEnabled?: boolean;
  siteCount?: number;
  workCategoryTypes?: IWorkCategoryTypes;
  buildings?: IBuilding[];
  sites?: ISite[];
}
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: site-collection.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: tag.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function TagFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    name: new FormControl(''),
    value: new FormControl(''),
    unitTags: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: tag.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { IUnitTag } from './unit-tag';

export interface ITag {
  id?: string;
  name?: string;
  value?: string;
  unitTags?: IUnitTag[];
}
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: tag.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: unit.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function UnitFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    number: new FormControl(''),
    unitTypeId: new FormControl(''),
    siteId: new FormControl(''),
    buildingId: new FormControl(''),
    isEnabled: new FormControl(''),
    building: new FormControl(''),
    site: new FormControl(''),
    unitType: new FormControl(''),
    unitTags: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: unit.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { IBuilding } from './building';
import { ISite } from './site';
import { IUnitType } from './unit-type';
import { IUnitTag } from './unit-tag';

export interface IUnit {
  id?: string;
  number?: string;
  unitTypeId?: string;
  siteId?: string;
  buildingId?: string;
  isEnabled?: boolean;
  building?: IBuilding;
  site?: ISite;
  unitType?: IUnitType;
  unitTags?: IUnitTag[];
}
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: unit.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: unit-tag.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function UnitTagFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    unitId: new FormControl(''),
    tagId: new FormControl(''),
    value: new FormControl(''),
    tag: new FormControl(''),
    unit: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: unit-tag.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { ITag } from './tag';
import { IUnit } from './unit';

export interface IUnitTag {
  id?: string;
  unitId?: string;
  tagId?: string;
  value?: string;
  tag?: ITag;
  unit?: IUnit;
}
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: unit-tag.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: unit-type.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function UnitTypeFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    code: new FormControl(''),
    description: new FormControl(''),
  });
};
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: unit-type.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export interface IUnitType {
  id?: string;
  code?: string;
  description?: string;
}
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: unit-type.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should generate files: work-category-types.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export interface IWorkCategoryTypes {
}
"
`;

exports[`Full Integration Tests MasterCorp Unit Service should not generate files 1`] = `Array []`;
